### ISSUE TITLE
Improve backfilling operation

### ISSUE THREAD

**Issue by GiedriusS** [prometheus/prometheus#11341]
## Proposal

If I am backfilling a bunch of recording rules over a period of 2w, for example, then `promtool tsdb create-blocks-from rules` creates a bunch of blocks. It will create a new block for every 1 minute. I think it should create bigger blocks with more data at once. Maybe there could even be an option `--block-length` that if set adds more data to one single block.

**Use case. Why is this important?**

I don't want to upload thousands of blocks to remote object storage in Thanos.

**Comment by yeya24** [comment]
I agree. I have tried rules backfilling before for about 2 month data. A lot of small blocks are created after the backfilling.

What I did is to start a Prometheus instance on that folder and enables allow overlapping blocks flag. Prometheus will wait a few seconds and compact the small blocks to larger ones.

**Comment by alexgreenbank** [comment]
Hello from the bug scrub.

Apologies for the delay on this.

@alexgreenbank to look into this based on recent experiences.

**Comment by aknuds1** [comment]
Is this now implemented in #8919 and #9511?

**Comment by GiedriusS** [comment]
I think so, let's close
